dex-it/src/test/scala/com/wavesplatform/it/sync/networking/NetworkIssuesTestSuite.scala

```diff
@@ -40,6 +43,41 @@ class NetworkIssuesTestSuite extends WsSuiteBase with HasToxiProxy {
     clearOrderBook()
   }
 
+  "DEXClient should place orders despite of short time disconnect from network" in {
+
+    val orders = (0 to 100).map { i =>
+      mkOrder(alice, wavesUsdPair, OrderType.SELL, 1.waves, 100 + i)
+    }
+
+    Await.ready(
+      for {
+        _ <- Future.successful(())
+
+        _ <- {
+          orders.foreach(dex1.asyncApi.place(_))
+          Future.successful()
+        }
+
+        _ <- {
+          Thread.sleep(500)
+          dex1.disconnectFromNetwork()
+          Thread.sleep(500)
+          dex1.connectToNetwork()
+          Future.successful()
+        }
+
+        orderBook <- dex1.asyncApi.orderBook(wavesUsdPair)
+
+      } yield {
+        orderBook.asks should have size 100
+        orders.foreach(dex1.asyncApi.waitForOrderStatus(_, OrderStatus.Accepted))
```

Comment:
All methods in `asyncApi` create a thread and work separately. This line actually runs 100 asynchronous methods `waitForOrderStatus`. All failures happen not in the thread of test. The right and simplest flow could be:
```scala
Await.result(
 // ...
)

orders.foreach(dex1.api.waitForOrderStatus(_, OrderStatus.Accepted)) // check statuses synchronously
```

Reply:
got it, thx